pkg/image/image.go

```diff
@@ -27,3 +28,19 @@ func NewBase(name string) Base {
 		name: name,
 	}
 }
+
+func GetCompressionPipeline(compression string, buildPipeline manifest.Build, inputPipeline manifest.FilePipeline) manifest.FilePipeline {
+	switch compression {
+	case "xz":
+		return manifest.NewXZ(buildPipeline, inputPipeline)
+	case "zstd":
+		return manifest.NewZstd(buildPipeline, inputPipeline)
+	case "gzip":
+		return manifest.NewGzip(buildPipeline, inputPipeline)
+	case "":
+		return inputPipeline
+	default:
+		// panic on unknown strings
+		panic(fmt.Sprintf("unsupported compression type %q", compression))
```

Comment:
(super nitpick) make this a `fmt.Errorf()` would prepare us already for a future with less panics :)